### PR TITLE
Bug 1951212:  User/Group details shows unrelated subjects in role bindings tab

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
/assign @spadgett 

Screen Shot 2021-04-27 at 10 48 09 AM

### CODE REVIEW

File: frontend/public/components/factory/table-filters.ts [hunk 1]
```diff
@@ -100,12 +100,7 @@ export const tableFilters: TableFilterMap = {
   'role-binding-roleRef-kind': (kind: string, binding) => binding.roleRef.kind === kind,
 
   // Filter role bindings by user name
-  'role-binding-user': (userName, binding) =>
-    _.some(binding.subjects, {
-      kind: 'User',
-      apiGroup: 'rbac.authorization.k8s.io',
-      name: userName,
-    }),
+  'role-binding-user': (userName, { subject }) => subject.name === userName,
```
Comment (florkbr) [inline frontend/public/components/factory/table-filters.ts:103]: @cyril-ui-developer do you know if `subjects` array in the original `binding` object will ever contain more than one array item? Wondering if we should display a binding that contains the `userName` in the array of `subjects` even if other subjects are present? (but not those bindings that do not contain the user name).

image

**Comment by florkbr** [comment]
/lgtm

Note - I cannot speak yet to what the intended design is here. Deferring to @spadgett on if this is the correct/intended behavior. 

## Before:
Screen Shot 2021-04-27 at 11 36 33 AM

## After:
Screen Shot 2021-04-27 at 11 36 47 AM

**Comment by cyril-ui-developer** [comment]
Group RoleBinding
Before:
Screen Shot 2021-04-27 at 10 48 09 AM

After:
Screen Shot 2021-04-27 at 3 04 20 PM

**Review by spadgett — APPROVED** [review]
/lgtm

Thanks @cyril-ui-developer. I think the new behavior is what was intended.

**Pull request merged**